tests/particles/insert_particle.cc

```diff
@@ -0,0 +1,85 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check that inserting particles works:
+// - in parallel when one processor is not inserting any particles
+// - and when all cells are locally owned on one processor
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/particles/generators.h>
+#include <deal.II/particles/particle_handler.h>
+
+#include "../tests.h"
+
+template <int dim, int spacedim>
+void
+test(const int refinement)
+{
+  {
+    parallel::distributed::Triangulation<dim, spacedim> tr(MPI_COMM_WORLD);
+
+    GridGenerator::hyper_cube(tr);
+    tr.refine_global(refinement);
+    MappingQ<dim, spacedim> mapping(1);
+
+    deallog << "Refinement: " << refinement << std::endl;
+
+    deallog << "Global active cells: " << tr.n_global_active_cells()
+            << std::endl;
+    deallog << "Locally owned active cells: "
+            << tr.n_locally_owned_active_cells() << std::endl;
+
+    Particles::ParticleHandler<dim, spacedim> particle_handler(tr, mapping);
+
+    std::vector<Point<dim>> positions(1, Point<dim>());
+    if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 1)
+      positions.clear();
+
+    particle_handler.insert_particles(positions);
+
+    deallog << "Global particles: " << particle_handler.n_global_particles()
+            << std::endl;
+    deallog << "Locally owned particles: "
+            << particle_handler.n_locally_owned_particles() << std::endl;
+  }
+
+  deallog << "OK" << std::endl;
+}
+
+
+
+int
+main(int argc, char *argv[])
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+
+  mpi_initlog();
```

Comment:
```suggestion
  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
  MPILogInitAll                    log;
```

I have just one suggestion: If you use the class `MPIInitLogAll` here (for example like in [tests/mpi/active_cell_index_01.cc](https://github.com/dealii/dealii/blob/master/tests/mpi/active_cell_index_01.cc)), output from each process will be logged in the output file. This way, you can also verify that the other process does not own any particle at all.

Reply:
I had that before, however @kronbichler suggested to print only output from rank 0. For me both would be fine. Please report back what you both prefer.

Reply:
Ah okay, I see his comment now, too :)

I thought it would be nice to check whether the other process does not own any process, to verify that the `ParticleHandler` members `number_of_locally_owned_particles` and `global_number_of_particles` hold the correct values. It would just add another sanity check, but I don't feel strongly about this.

Reply:
Yes, I agree. It's useful to see what the other processes have.

Reply:
I see we lean more towards the other options and I obviously don't disagree then. I would just add that you could see that other processes wouldn't own anything because we had 1 global particle and 1 that is locally owned - unless we don't have other bugs, that excludes the other process from having a particle. (I just speak from experience that after updating hundreds of tests that I prefer the compact output that contains all info.)

Reply:
And @slfuchs my apologies for getting you into this round-trip in your first test contribution, but the additional things checked here can make sense.

Reply:
No worries, as long as we reach a converged state that's fine :smile: